jvm/src/main/java/com/microsoft/bond/protocol/FastBinaryReader.java

```diff
@@ -0,0 +1,147 @@
+package com.microsoft.bond.protocol;
+
+import com.microsoft.bond.BondDataType;
+
+import java.io.InputStream;
+
+public class FastBinaryReader<S extends InputStream> implements TaggedProtocolReader {
+    public final S stream;
+    private final short version;
+    private final BinaryStreamReader<S> reader;
+
+    public FastBinaryReader(final S stream, final short version) {
+        this.stream = stream;
+        this.version = version;
+        this.reader = new BinaryStreamReader<S>(stream);
+    }
+
+    private BondDataType readType() {
+        return new BondDataType(readInt8());
+    }
+
+    @Override
+    public void readStructBegin() {}
+
+    @Override
+    public void readBaseBegin() {}
+
+    @Override
+    public void readStructEnd() {
+        // FIXME: Verify that this is BT_STOP.
```

Comment:
When we do all the other checking I skipped. Just getting the happy path working so you can run deserialization at all.